godot/ModularWeapons/ModularWeapon/src/Projectile/Basic/BasicProjectile.gd

```diff
@@ -1,23 +1,22 @@
 extends Projectile
 
-onready var timer: Timer = $Timer
-onready var tween := $Tween
+@onready var timer: Timer = $Timer
 
 
 # Called after the base _setup to allow for specialized configuration.
 func _post_setup() -> void:
 	if not is_inside_tree():
-		yield(self, "ready")
+		await self.ready
 	timer.start(lifetime)
 
 
 func _physics_process(delta: float) -> void:
 	var movement := _update_movement(delta)
 
-	var collision := move_and_collide(movement)
+	var collision : KinematicCollision2D= move_and_collide(movement)
 
 	if collision:
-		emit_signal("collided", collision.collider, collision.position)
+		emit_signal("collided", collision.get_collider(), collision.get_position())
```

Comment:
Always prefer the types instead of string, in this case we use `Signal` directly: `collided.emit(collision.get_collider(), collision.get_position())`